sklearn/metrics/_regression.py

```diff
@@ -387,7 +394,11 @@ def median_absolute_error(y_true, y_pred, *, multioutput='uniform_average'):
     """
     y_type, y_true, y_pred, multioutput = _check_reg_targets(
         y_true, y_pred, multioutput)
-    output_errors = np.median(np.abs(y_pred - y_true), axis=0)
+    if sample_weight is None:
+        output_errors = np.median(np.abs(y_pred - y_true), axis=0)
+    else:
+        output_errors = _weighted_percentile(np.abs(y_pred - y_true),
```

Comment:
since `sample_weight` can be a list at this point, you will need to validate it.

```python
from ..utils.validation import _check_sample_weight
sample_weight = _check_sample_weight(sample_weight, y_pred)
```